PurchasesCoreSwift/Identity/PurchaserInfoManager.swift

```diff
@@ -0,0 +1,191 @@
+        self.deviceCache = deviceCache
+        self.backend = backend
+        self.systemInfo = systemInfo
+    }
+
+    @objc public func fetchAndCachePurchaserInfo(appUserID: String,
+                                                 isAppBackgrounded: Bool,
+                                                 completion: ReceivePurchaserInfoBlock?) {
+        deviceCache.setPurchaserInfoCacheTimestampToNow(appUserID: appUserID)
+        operationDispatcher.dispatchOnWorkerThread(withRandomDelay: isAppBackgrounded) {
+            self.backend.getSubscriberData(appUserID: appUserID) { purchaserInfo, error in
+                if error != nil {
+                    self.deviceCache.clearPurchaserInfoCacheTimestamp(appUserID: appUserID)
+                } else if let info = purchaserInfo {
+                    self.cache(purchaserInfo: info, appUserID: appUserID)
+                }
+
+                if let completion = completion {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completion(purchaserInfo, error)
+                    }
+                }
+
+            }
+        }
+    }
+
+    @objc public func fetchAndCachePurchaserInfoIfStale(appUserID: String,
+                                                        isAppBackgrounded: Bool,
+                                                        completion: ReceivePurchaserInfoBlock?) {
+        let cachedPurchaserInfo = cachedPurchaserInfo(appUserID: appUserID)
+        let isCacheStale = deviceCache.isPurchaserInfoCacheStale(appUserID: appUserID,
+                                                                 isAppBackgrounded: isAppBackgrounded)
+        let needsToRefresh = isCacheStale || cachedPurchaserInfo == nil
+        if needsToRefresh {
+            Logger.debug(isAppBackgrounded
+                            ? Strings.purchaserInfo.purchaserinfo_stale_updating_in_background
+                            : Strings.purchaserInfo.purchaserinfo_stale_updating_in_foreground)
+            fetchAndCachePurchaserInfo(appUserID: appUserID,
+                                       isAppBackgrounded: isAppBackgrounded,
+                                       completion: completion)
+            Logger.rcSuccess(Strings.purchaserInfo.purchaserinfo_updated_from_network)
+        } else {
+            if let completion = completion {
+                operationDispatcher.dispatchOnMainThread {
+                    completion(cachedPurchaserInfo, nil)
+                }
+            }
+        }
+    }
+
+    @objc(sendCachedPurchaserInfoIfAvailableForAppUserID:)
+    public func sendCachedPurchaserInfoIfAvailable(appUserID: String) {
+        guard let info = cachedPurchaserInfo(appUserID: appUserID) else {
+            return
+        }
+
+        sendUpdateIfChanged(purchaserInfo: info)
+    }
+
+    @objc(purchaserInfoWithAppUserID:completionBlock:)
+    public func purchaserInfo(appUserID: String, completionBlock: ReceivePurchaserInfoBlock?) {
```

Comment:
```suggestion
    public func purchaserInfo(appUserID: String, completion maybeCompletion: ReceivePurchaserInfoBlock?) {
```

we have a mix of `completion` and `completionBlock` in our codebase, perhaps we should standardize?

Reply:
Agreed! `completion` is it 😄